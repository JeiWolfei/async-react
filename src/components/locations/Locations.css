.locations {
  width: 31%;
  float: right;
  background-image: repeating-linear-gradient(orange,yellow, pink, orange, pink);
  h2 {
    padding-left: 30px;
  }
  @media screen and (max-width: 536px) {
    width: 100%;
  }
}
