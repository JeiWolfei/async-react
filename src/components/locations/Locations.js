import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Location from './Location';
import { getLocations } from '../../services/rickAndMortyApi';
import { withPaging } from '../paging/Paging';
import styles from './Locations.css';

class Locations extends PureComponent {
  static propTypes = {
    page: PropTypes.number,
    updateTotalPages: PropTypes.func.isRequired
  };

  static defaultProps = {
    page: 1
  }

  state = {
    locations: []
  }

  fetchLocations = () => {
    getLocations(this.props.page)
      .then(response => {
        this.props.updateTotalPages(response.totalPages);
        this.setState({ locations: response.results });
      });
  }

  componentDidMount() {
    this.fetchLocations();
  }

  componentDidUpdate(prevProps) {
    if(prevProps.page !== this.props.page) {
      this.fetchLocations();
    }
  }

  render() {
    const locations = this.state.locations.map(loc => {
      return (
        <Location key={loc.id} location={loc}/>
      );
    });
    return (
      <div className={styles.locations}>
        <h2>Locations</h2>
        {locations}
      </div>
    );
  }
}

export default withPaging(Locations);

