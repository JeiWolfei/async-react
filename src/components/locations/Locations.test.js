import React from 'react';
import renderer from 'react-test-renderer';
import Locations from './Locations';

jest.mock('../../services/rickAndMortyApi.js');

describe('Locations', () => {
  it('matches a snapshot', () => {
    const tree = renderer.create(
      <Locations updateTotalPages={jest.fn()}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
