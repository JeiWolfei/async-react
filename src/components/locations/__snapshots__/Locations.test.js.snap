// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Locations matches a snapshot 1`] = `
Array [
  <section
    style={
      Object {
        "background": "cyan",
        "marginLeft": "10px",
        "textAlign": "center",
        "zIndex": "auto",
      }
    }
  >
    <div
      style={
        Object {
          "fontSize": "12px",
          "margin": "10px 5px 0px 5px",
        }
      }
    >
      <p>
        1
         of 
        1
      </p>
    </div>
  </section>,
  <div
    className="locations"
  >
    <h2>
      Locations
    </h2>
  </div>,
]
`;
